cmds/boot/systemboot/main.go

```diff
@@ -178,6 +178,17 @@ func runIPMICommands() {
 				log.Printf("Failed to get the information of Processors: %v.", err)
 			}
 
+			BootDriveInfo, err := ocp.GetOemIpmiBootDriveInfo(si)
+			if err == nil && BootDriveInfo != nil {
```

Comment:
if err != nil {
}
if BootDriveInfo != nil {
}
As it is, the code does not distinguish the error case from the no info case, and I think that's an important difference.

Reply:
Done